### DISCUSSION TITLE
Authorized API

### DISCUSSION THREAD

**Discussion by mesfandi** [elsa-workflows/elsa-core#2822] [Q&A]
Hi,
I know that I can use Elsa plugin to add Bearer to the header. Could you please let me know how I can Authorize the httpendpoints created by ELSA? I want a workflow only run for the authorized user who has the API key.
I have blazor web assembly with asp.net server and identity server for authentication. However, it seems that the httpendpoints from Elsa do not come to the asp.net pipeline I defined. 
I created a hello world sample in the Elsa and enabled httpendpoint Authorize  and left the policy blank (see attached). Still, my hello api work without token.   
![image](https://user-images.githubusercontent.com/40899452/157682784-a68c6fcb-802b-4b4c-a649-bf06ce25f895.png)
Do you have any example for Authorized HTTP endpoints?

**Comment by sfmskywalker** [comment]
Hi @mesfandi 

I don't think there's a public example available yet, but the key is to install the `AuthenticationBasedHttpEndpointAuthorizationHandler`, which you can do from Startup like this:

```csharp
services.AddSingleton<AuthenticationBasedHttpEndpointAuthorizationHandler>();

services.AddElsa(elsa => elsa.AddHttpActivities(http => http.HttpEndpointAuthorizationHandlerFactory = sp => sp.GetRequiredService<AuthenticationBasedHttpEndpointAuthorizationHandler>()));
```

**Comment by dicksonding** [comment]
Hi @sfmskywalker,

I have managed to get the authorization running for HttpEndpoint, however, is there any way I could perform Authorization for EventReceived / Signal? As both EventReceived / Signal can be trigger through http but no settings for policy.
